ledger/daml-on-sql/src/main/scala/on/sql/Main.scala

```diff
@@ -17,5 +17,12 @@ object Main {
   }
 
   private[sql] def run(config: SandboxConfig): Unit =
-    new ProgramResource(SandboxServer.owner(Name, config)).run(ResourceContext.apply)
+    config.sqlStartMode.foreach {
+      case PostgresStartupMode.MigrateOnly =>
+        new ProgramResource(SandboxServer.migrateOnly(config)).run(ResourceContext.apply)
+        sys.exit()
```

Comment:
You don't wanna do this. `ProgramResource#run` doesn't block.

Instead, make `migrateOnly` return a `Future[Unit]`, and then do this:

```suggestion
        SandboxServer.migrateOnly(config)
          .onComplete {
             case Success(_) =>
               sys.exit(0)
             case Failure(exception) =>
               System.err.println(exception.getMessage)
               sys.exit(1)
           }
```

Reply:
resolved in latest push with move to Future